homeassistant/components/nws/weather.py

```diff
@@ -273,6 +306,45 @@ def _forecast(
             forecast.append(data)
         return forecast
 
+    @overload
+    def _forecast_extra(
+        self,
+        nws_forecast: None,
+        mode: str,
+    ) -> None: ...
+
+    @overload
+    def _forecast_extra(
+        self,
+        nws_forecast: list[dict[str, Any]],
+        mode: str,
+    ) -> list[ExtraForecast]: ...
+
+    def _forecast_extra(
```

Comment:
The overloads seems unnecessary and it could return an empty list instead of `None` by default I think.

Reply:
This function is intended to mirror the normal forecast data, so that it is easier to maintain the code here.  Can we return `[]` for `_forecast` as well?

Reply:
I can't simply remove the typing overload, mypy complains.  So I will wait on feedback on whether it is acceptable to return an empty list for `_forecast`.

Reply:
I can't figure out how to get mypy happy here.  For some reason the only solution is to use cast or these overloads.  I made the change return `[]` for both cases, but this now requires a cast to `JsonValueType` to make mypy happy.

Reply:
So I took a bit of closer look to this.
`self.nws.forecast` and `self.nws.forecast_hourly` is typed in the library that it can return `None` while it will always return in minimum an empty list so actually that typing should be updated in the library which would also simplify here as we don't need to consider a `None` return value for the forecast properties in the lib.

Besides that we had some issue with typing also previously with `JsonValueType` so casting to make mypy happy here I don't see an issue with.
All these methods can in minimum return an empty list and that should be fine as it's the default for the service anyhow.

Reply:
> So I took a bit of closer look to this.
self.nws.forecast and self.nws.forecast_hourly is typed in the library that it can return None while it will always return in minimum an empty list

You are correct!  This was changed in the upstream library some time ago, but typing didn't keep up.  When that happens, we can simplify here when that is fixed.